src/agents/memory/async_redis_checkpointer.py

```diff
@@ -528,3 +529,13 @@ async def alist_llm_usage_records(self, cluster_id: str, ttl: int) -> list[dict]
                 latest_keys.append(key)
         records = await self.conn.mget(latest_keys)
         return [json.loads(record) for record in records if record]
+
+
+def get_async_redis_saver() -> AsyncRedisSaver:
+    """
+    Returns an instance of AsyncRedisSaver with a async redis connection as defined in the config files.
+    """
+    connection = get_redis().connection
+    if connection is None:
+        raise ValueError("Failed to retrieve a valid Redis connection.")
+    return AsyncRedisSaver(connection)
```

Comment:
Reference to this [comment](https://github.com/kyma-project/kyma-companion/pull/508/files#r2073289957):

```
def get_async_redis_saver() -> AsyncRedisSaver:
    """
    Returns an instance of AsyncRedisSaver with a async redis connection as defined in the config files.
    """
    return AsyncRedisSaver(Redis().get_connection())
```

IMO, all these checks should be happening in the Redis class.

Reply:
Valid point.